### PR TITLE
Enable configuration caching

### PR DESCRIPTION
**Pull request by clayburn**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -126,20 +127,21 @@ val copyMavenScripts = tasks.register<Copy>("copyMavenScripts") {
     group = "build"
     description = "Copies the Maven source and generated scripts to output directory."
 
-    inputs.property("project.version", appVersion)
+    val projectVersion = appVersion
```
Comment (clayburn) [inline build.gradle.kts:130]: Note: had to copy this into a local variable in order to maintain configuration caching.

File: build.gradle.kts [hunk 2]
```diff
@@ -93,6 +93,7 @@ val copyGradleScripts = tasks.register<Copy>("copyGradleScripts") {
     group = "build"
     description = "Copies the Gradle source and generated scripts to output directory."
 
+    val projectVersion = appVersion
     inputs.property("project.version", appVersion)
```
Comment (etiennestuder) [inline build.gradle.kts:97]: I would use `projectVersion` here instead of `appVersion`.

File: build.gradle.kts [hunk 3]
```diff
@@ -260,6 +262,11 @@ tasks.register<CreateGitTag>("createReleaseTag") {
 
 tasks.named("githubRelease") {
     dependsOn("createReleaseTag")
+    notCompatibleWithConfigurationCache("$name task does not support configuration caching")
```
Comment (etiennestuder) [inline build.gradle.kts:265]: Same comment as on the CCUD plugin.

**Review by etiennestuder — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (etiennestuder) [reply build.gradle.kts:130]: I suggest to add this as a comment, e.g.

```
    val projectVersion = appVersion. // local variable required for configuration cache compatibility
```

Please do the same for Maven further down, too.

**Pull request merged**